### PR TITLE
Use retagged trivy-db image

### PR DESCRIPTION
**Pull request by fhielpos**
Towards: https://github.com/giantswarm/giantswarm/issues/31965

### CODE REVIEW

File: helm/trivy/values.yaml [hunk 1]
```diff
@@ -37,3 +37,6 @@ trivy:
     privileged: false
     readOnlyRootFilesystem: true
     allowPrivilegeEscalation: false
+  
+  trivy:
+    dbRepository: gsoci.azurecr.io/giantswarm/trivy-db
```
Comment (stone-z) [inline helm/trivy/values.yaml:42]: Same Aliyun question here

Comment (fhielpos) [reply helm/trivy/values.yaml:42]: Yes but here we don't have it split in registry and repository. Will check what we can do

Comment (fhielpos) [reply helm/trivy/values.yaml:42]: Unfortunately here there is now way for us to split this template into two, so we need to overwrite the value directly at the catalog level

**Review by stone-z — APPROVED** [review]
Scans are already not pulling from aliyun, so this at least isn't a regression. I'm fine to do it at the catalog level if you want, or let eventual zot be the solution here.

**Pull request merged**